pkg/daemon/on_disk_validation.go

```diff
@@ -5,18 +5,119 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"reflect"
+	"strings"
 
 	ign2types "github.com/coreos/ignition/config/v2_2/types"
 	ign3types "github.com/coreos/ignition/v2/config/v3_4/types"
+	"github.com/coreos/vcontext/path"
 	"github.com/google/go-cmp/cmp"
 	mcfgv1 "github.com/openshift/api/machineconfiguration/v1"
 	ctrlcommon "github.com/openshift/machine-config-operator/pkg/controller/common"
+	"github.com/openshift/machine-config-operator/pkg/daemon/constants"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/klog/v2"
 )
 
+// Match the file captured in the event -> correspoding MC spec
+func validateOnDiskStateForEvent(currentConfig *mcfgv1.MachineConfig, rootPath string, path path.ContextPath) error {
+	ignconfigi, err := ctrlcommon.IgnParseWrapper(currentConfig.Spec.Config.Raw)
+	if err != nil {
+		return fmt.Errorf("failed to parse Ignition for validation: %w", err)
+	}
+	switch typedConfig := ignconfigi.(type) {
+	case ign3types.Config:
+		value, err := getValueInMCForPath(ignconfigi.(ign3types.Config), path)
+		if err != nil {
+			return fmt.Errorf("failed to get valye: %w", err)
+		}
+		switch path.Tag {
+		case "passwd":
+			v, ok := value.([]ign3types.SSHAuthorizedKey)
+			if !ok {
+				return fmt.Errorf("failed to convert")
```

Comment:
suggestion: For each of these, include what it failed to convert to, e.g., `fmt.Errorf("failed to convert to SSHAuthorizedKey"`). That will make debugging much easier.

Reply:
Agreed